kong/plugins/ratelimiting/access.lua

```diff
@@ -13,29 +14,50 @@ function _M.execute(conf)
   else
     identifier = ngx.var.remote_addr
   end
-
-  -- Load current metric for configured period
-  local current_metric, err = dao.ratelimiting_metrics:find_one(ngx.ctx.api.id, identifier, current_timestamp, conf.period)
-  if err then
-    ngx.log(ngx.ERR, tostring(err))
-    utils.show_error(500)
+  local periods_and_limits = {}
+  if conf.limit then
+    for _,v in ipairs(conf.limit) do
+      local parts = stringy.split(v, ":")
+      periods_and_limits[parts[1]] = parts[2]
+    end
   end
 
-  -- What is the current usage for the configured period?
-  local current_usage
-  if current_metric ~= nil then
-    current_usage = current_metric.value
-  else
-    current_usage = 0
-  end
+  local least_remaining_limit = {}
+  least_remaining_limit["remaining"] = 999999999 -- A huge number so that any limit will be actually less than this
 
-  local remaining = conf.limit - current_usage
-  ngx.header[constants.HEADERS.RATELIMIT_LIMIT] = conf.limit
-  ngx.header[constants.HEADERS.RATELIMIT_REMAINING] = math.max(0, remaining - 1) -- -1 for this current request
+  for period, limit in pairs(periods_and_limits) do
+    limit = tonumber(limit)
+    -- Load current metric for configured period
+    local current_metric, err = dao.ratelimiting_metrics:find_one(ngx.ctx.api.id, identifier, current_timestamp, period)
+    if err then
+      ngx.log(ngx.ERR, tostring(err))
+      utils.show_error(500)
```

Comment:
This is not available anymore, please rebase and use the new methods in #221.

I am also really not a fan of performing a SELECT in a loop. This could maybe be done in a simple SELECT call.